### PR TITLE
Add batch processing to 'draft' action, add test artifacts for zero, …

### PR DESCRIPTION
**Pull request by qchaupds**
Pull request to close DOI Batch Processing capability #25

Batch processing allow to process series of PDS4 labels.

### CODE REVIEW

File: pds_doi_core/actions/draft.py [hunk 1]
```diff
@@ -83,30 +148,39 @@ def run(self,
...
-                                                                         submitter,
-                                                                         [doi_fields],
-                                                                         output_content=o_doi_label)
+            # Transform the PDS4 label to an OSTI record.
+            doi_label = self._transform_pds4_label_into_osti_record(input_file,node,submitter,contributor_value)
 
-        # Write a transaction for the 'reserve' action.
-        transaction_obj.log()
+            # The returned label can be None if it is not an expected XML file.
+            if doi_label:
```
Comment (tloubrieu-jpl) [inline pds_doi_core/actions/draft.py:174]: This test looks like an implicit error management. That would be more readable to have an explicit exception (invalid PDS4Input...) raised by the method _transform_pds4_label_into_osti_record().
But 
I don't know what case would raise this exception. Do you know what case can make the doi_abel == None ?

File: pds_doi_core/actions/draft.py [hunk 2]
```diff
@@ -46,6 +50,69 @@ def add_to_subparser(cls, subparsers):
                                    default='osti',
                                    metavar='osti')
 
+    def _resolve_input_into_list_of_names(self, input):
+        """Function receives an input which can be one name or a list of names separated by a comma or a directory.  Function will return a list of names.
+        """
+        o_list_of_names = []
+        if ',' in input:
```
[the remark this answers is not in the document]
Comment (qchaupds) [reply pds_doi_core/actions/draft.py:57]: You are right.  I started with one train of thought and kept going.
Splitting using the comma ',' right away allow you to use the for loop to check if the token is a directory or a file.  
Code is much simpler now.

File: pds_doi_core/actions/draft.py [hunk 3]
```diff
@@ -46,6 +50,69 @@ def add_to_subparser(cls, subparsers):
...
+            # Get names separated by comma.
+            split_tokens  = input.split(',')
+            for token in split_tokens:
+                # Only save the file name if it is not an empty string as in the case of a comma being the last character:
+                #    -i https://pds-imaging.jpl.nasa.gov/data/nsyt/insight_cameras/data/collection_data.xml,
+                # or no name provided with just a comma:
+                #    -i ,
+                if len(token) > 0:
+                    o_list_of_names.append(token)
+        elif os.path.isdir(input):
```
[the remark this answers is not in the document]
Comment (qchaupds) [reply pds_doi_core/actions/draft.py:67]: Putting the elif and the following code inside the for loop works now.

File: pds_doi_core/actions/draft.py [hunk 4]
```diff
@@ -83,30 +148,39 @@ def run(self,
...
-        if not input.startswith('http'):
-            xml_tree = etree.parse(input)
-        else:
-            response = requests.get(input)
-            xml_tree = etree.fromstring(response.content)
+        # The value of input can be a list of names, or a directory.  Resolve that to a list of names.
+        list_of_names = self._resolve_input_into_list_of_names(input)
+        # Return immediately if there are no files found.
+        # Note that exit() function is not used as it would interfere with the unit test.
+        if len(list_of_names) == 0:
```
[the remark this answers is not in the document]
Comment (qchaupds) [reply pds_doi_core/actions/draft.py:155]: Yes, it can happen if the input for the -i is a directory and there is nothing inside.  As it turns out, we don't need to do the check since the while loop will take care of that (iterating over a list of zero). 

I removed len() check and the code still runs OK.  You are right.  There's no need to do an additional check when the for loop works just fine.

File: pds_doi_core/actions/draft.py (hunk 1 above)
Comment (qchaupds) [reply pds_doi_core/actions/draft.py:174]: To avoid having to check for doi_label be None, the function _transform_pds4_label_into_osti_record() can return an empty tree (which is not None).  There's no need to check for None-ness.

The code is much simpler now.
